packages/server/lib/plugins/child/run_require_async_child.js

```diff
@@ -111,14 +111,14 @@ function run (ipc, file, projectRoot) {
     try {
       debug('Trying to use esbuild to run their config file.')
       // We prefer doing this because it supports TypeScript files
-      require.resolve('esbuild')
+      require.resolve('esbuild', { paths: [process.cwd()] })
```

Comment:
The absence of this line was actually a bug and it was only working on my machine because of some GLOBAL transitive deps with `esbuild`.

To debug this, you can add

```js
require.resolve.paths('esbuild')
```

to a production-built binary installed in a real project. We should really consider moving some of these low-dependency or no-dependency tests OUTSIDE of the system-tests and into the `binary-system-tests` phase.